### PR TITLE
Add list commands to project, cloud profile, dataset

### PR DESCRIPTION
**Pull request by ExtinctWolf83**

### CODE REVIEW

File: deploifai/cloud_profile/list.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+)
+from deploifai.api import DeploifaiAPIError
+
+
+@click.command("list")
+@pass_deploifai_context_obj
+@is_authenticated
+def list_profile(context: DeploifaiContextObj):
+    """
+    Command to list all cloud profiles linked to the current workspace
```
Comment (98sean98) [inline deploifai/cloud_profile/list.py:15]: I think we can shorten the text a bit more with just `List all cloud profiles in the current workspace`

File: deploifai/data/list.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    is_authenticated,
+    project_found,
+)
+from deploifai.api import DeploifaiAPIError
+
+
+@click.command("list")
+@pass_deploifai_context_obj
+@is_authenticated
+def list_data(context: DeploifaiContextObj):
```
Comment (98sean98) [inline deploifai/data/list.py:14]: As discussed, listing priority should be
1. command option for project name
2. project id in local config (if found)
3. current workspace in global config (if project not found in local config)

**Review by 98sean98 — COMMENTED** [review]
Still missing some necessary changes using the `whereProject` argument queries for data storages querying in the `dataset list` command

**Review by 98sean98 — CHANGES_REQUESTED** [review]
Just some small changes.

File: deploifai/server/list.py [hunk 3]
```diff
@@ -15,8 +17,45 @@ def list_server(context: DeploifaiContextObj):
...
+                fg="red",
+            )
+            return
+        if len(projects_data) == 0:
+            click.secho(
+                f"Project \"{project}\" cannot be found in workspace \"{current_workspace}\"",
+                fg="red",
+            )
+            return
+        variables = {"project": {"is": {"name": {"equals": project}}}}
```
[the remark this answers is not in the document]
Comment (ExtinctWolf83) [reply deploifai/server/list.py]: refactored

File: deploifai/api/__init__.py [hunk 4]
```diff
@@ -91,16 +91,12 @@ def get_containers(self, data_storage_id):
         except TypeError:
             return []
 
-    def get_data_storages(self, workspace_id):
+    def get_data_storages(self, workspace: str, where_data_storage: str = None):
```
[the remark this answers is not in the document]
Comment (ExtinctWolf83) [reply deploifai/api/__init__.py:94]: refactored

File: deploifai/api/__init__.py [hunk 5]
```diff
@@ -558,26 +560,35 @@ def create_training_server(self, name: str, project_id: str, cloud_profile_id: s
         except KeyError:
             raise DeploifaiAPIError("Could not create Training Server. Please try again.")
 
-    def get_training_server(self, workspace: str):
+    def get_training_server(self, workspace: str, where_training_server: str = None):
```
[the remark this answers is not in the document]
Comment (ExtinctWolf83) [reply deploifai/api/__init__.py:563]: refactored

File: deploifai/server/list.py [hunk 6]
```diff
@@ -1,22 +1,78 @@
...
-    click.echo("Training Server List:")
+    elif "id" in context.local_config["PROJECT"]:
+        project_id = context.local_config["PROJECT"]["id"]
+        # query for workspace name from api
+        fragment = """
+                        fragment project on Project {
+                            name
+                            account{
+                                username
+                            }
```
[the remark this answers is not in the document]
Comment (ExtinctWolf83) [reply deploifai/server/list.py:55]: yes
removed

**Review by 98sean98 — APPROVED** [review]

**Pull request merged**